pontoon/base/models/entity.py

```diff
@@ -871,19 +884,38 @@ def for_project_locale(
             # Search in source strings
             if not search_translations_only:
                 # Search in string (context) identifiers
-
-                q_key = Q()
-                # TODO: Uncomment the 5 lines below to reactivate the
-                #       context identifiers filter once .ftl bug is fixed (issue #3284):
-                # q_key = (
-                #     Q(**{f"key__{case_lookup}": (search)})
-                #     if search_identifiers
-                #     else Q()
-                # )
+                case_lookup = "contains" if search_match_case else "icontains"
+                q_key = (
+                    Q(**{f"key__{case_lookup}": (search)})
+                    if search_identifiers
+                    else Q()
+                )
 
                 entity_filters = (
-                    Q(**{f"string__{i}regex": rf"{y}{s}{y}"})
-                    | Q(**{f"string_plural__{i}regex": rf"{y}{s}{y}"})
+                    Q(
+                        Q(resource__format="ftl")
+                        & (
+                            Q(**{f"string__{i}regex": rf"=.*{y}{s}{y}.*"})
+                            if not search_identifiers
+                            else Q(**{f"string_plural__{i}regex": rf"{y}{s}{y}"})
+                        )
+                    )
+                    | Q(
+                        ~Q(resource__format="ftl")
+                        & Q(**{f"string__{i}regex": rf"{y}{s}{y}"})
+                    )
+                    | Q(
+                        Q(resource__format="ftl")
+                        & (
+                            Q(**{f"string_plural__{i}regex": rf"=.*{y}{s}{y}.*"})
+                            if not search_identifiers
+                            else Q(**{f"string_plural__{i}regex": rf"{y}{s}{y}"})
+                        )
```

Comment:
`Entity.string_plural` is only used for Gettext (.po) strings (I know... lame... but we'll refactor that within the next 6 months), so we don't need this code.